Add clarifications about git in README

DimitarVanguelov opened the pull request:
Per discussion with @pankgeorg on Slack, opening this PR to document the issue described below. Per his request, I'm adding it as a part of a new FAQ section. Unfortunately, not sure what else to put in here at the moment, so perhaps we can have this as a living PR until the FAQ section gets sufficiently fleshed out.

**The original issue**: our git platform (CodeCommit) requires re-authentication several times per 24-hour period. This is a problem when PlutoSliderServer is watching a different repo than the one holding the code for the server as it crashes the Julia git client (even though it keeps the systemd service going). The result is that no new notebooks or changes get picked up the service.

**The workaround** is to set the `JULIA_PKG_USE_CLI_GIT` environment variable to `true` (thanks to @alecloudenback for suggesting this fix).

alecloudenback commented:
I would maybe phrase it more generally: ie that if your git installation or remote requires special configuration or authentication, then try using CLI git instead of bundled.

fonsp commented:
Thanks!

fonsp commented:
Maybe we should recommend this to everyone and add the line `export JULIA_PKG_USE_CLI_GIT=true` in the tutorial. Is there any reason not to use it? @alecloudenback

pankgeorg commented:
> Maybe we should recommend this to everyone and add the line `export JULIA_PKG_USE_CLI_GIT=true` in the tutorial. Is there any reason not to use it? @alecloudenback

let's use it all the time, LibGit2 has a number of issues the CLI git doesn't. The only reason _not_ to use it is if the underlying machine has no git installed at all. We should probably note that users need to make sure the `git` command works on the host machine.

alecloudenback commented:
I don't know the full history, but if git started being bundled with Julia because CLI git had issues, that suggests there's some solid reason for having bundled git. 

I think the issues @DimitarVanguelov first mentioned were due to non-standard git configuration settings in our environment, so may not be an issue for (most?) other users.

pankgeorg commented:
https://github.com/JuliaLang/Pkg.jl/issues/2679

This gives some insight.

DimitarVanguelov commented:
@pankgeorg thanks, very informative discussion in that thread. 

I can amend the PR and take out the FAQ section. My initial idea was just to have a sentence or two using git CLI, not an FAQ (I certainly don't have enough material for one, though I'm sure one would be good to have eventually).

fonsp commented:
Alright, let's go with setting the ENV variable for all users, @DimitarVanguelov can you update the PR? Thanks everyone for the input!

DimitarVanguelov commented:
I updated the PR, but please don't merge yet:
- I put the env var setting in the startup script, just above the `julia` invocation, but I need to double check how we did this at work (I will check tomorrow). We might have put it in the step 1 script, but his seems like a more natural place to set it, so I welcome input here.
- I also deleted any trailing white spaces from the README, hope that's ok.
- I also updated the Julia versions in step 1 to be 1.8.0. 
I can back the last two things out if you think they're not appropriate for this PR.

DimitarVanguelov commented:
Also, maybe using [Git.jl](https://github.com/JuliaVersionControl/Git.jl) is worth exploring in a future PR?

DimitarVanguelov commented:
Sorry for the delay, been a bit busy for me lately. The way my co-worker had set it up is to put the `JULIA_PKG_USE_CLI_GIT` env variable in a script under `/etc/profile.d`, which is either good form or overkill for this sort of thing, I just don't know.

pankgeorg commented on README.md:
```diff
@@ -215,13 +215,17 @@ touch pluto-slider-server-environment/PlutoDeployment.toml
+`run_git_directory` will periodically call `git pull`, which requires the `start_dir` to be a repository in which you can `git pull` without password (which means it's either public, or you have the required keys in `~/.ssh/` and your git's provider security page!)
+
+>**Note**
+>Julia by default uses `libgit2` for git operations, but that has at times proven problematic, especially in cloud environments like AWS's CodeCommit
+>where re-authentication is required at regular intervals. It is thus advisable to set the `JULIA_PKG_USE_CLI_GIT` environment variable to `true`.
```
```suggestion
>Julia by default uses `libgit2` for git operations, [which can be problematic](https://github.com/JuliaLang/Pkg.jl/issues/2679). It is also known to cause issues  in cloud environments like AWS's CodeCommit
>where re-authentication is required at regular intervals. 
> 
> A simple workaround is to set the `JULIA_PKG_USE_CLI_GIT` environment variable to `true`, which will fallback to the system git (the one on the shell).
> Make sure that this is installed! (`sudo apt-get install git` does the trick in Ubuntu).

Also note that git pull may fail if you force push the repository!
```

pankgeorg commented:
Sorry this took quite some time! I added some more general notes and I'm merging this, thanks for contributing!
